leetcode/0001/TwoSum.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections.Generic;
+
+public class Solution
+{
+    public int[] TwoSum(int[] nums, int target)
+    {
+        return SolveWithOnePass(nums, target);
+    }
+
+    public int[] SolveWithOnePass(int[] nums, int target)
+    {
+        var set = new Dictionary<int, int>(nums.length);
+
+        for(int x = 0; x < nums.Length; x++)
+        {
+            if (set.TryGetValue(nums[x], out var result))
+            {
+                if (x == result) continue;
+
+                return new int[] { x, result };
+            }
+            
+            set[target-nums[x]] = x;
```

Comment:
If this was added to the set before the test, then we fall into the problem where multiple of the same value in `nums` would cause a problem.  For example "[3,3,] target=6"